### PR TITLE
k8s,node: Reuse retrieveNodeInformation to retrieve node labels

### PR DESCRIPTION
**Pull request by pchaigno**
When creating the host endpoint, we need to retrieve existing node labels. Instead of making a new API call, we can reuse `retrieveNodeInformation()` to get those labels.

@tgraf In #11543, [you mentioned](https://github.com/cilium/cilium/pull/11543#discussion_r426408765) it would be best to reuse `retrieveNodeInformation()`. Is this what you had in mind?

Would it be worth adding these labels to `CiliumNode` as well?

### CODE REVIEW

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-me-please

File: pkg/node/node_labels.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 Authors of Cilium
```
Comment (aanm) [inline pkg/node/node_labels.go]: Small nit the file could be called `labels.go`

**Review by aanm — APPROVED** [review]

File: pkg/endpointmanager/manager.go [hunk 2]
```diff
@@ -479,16 +479,10 @@ func (mgr *EndpointManager) AddHostEndpoint(ctx context.Context, owner regenerat
 	epLabels := labels.Labels{}
 	epLabels.MergeLabels(labels.LabelHost)
 
-	if k8s.IsEnabled() {
```
Comment (nebril) [inline pkg/endpointmanager/manager.go:482]: Are we sure we want to get rid of this check? I guess we don't run outside of k8s anyway.

**Review by nebril — APPROVED** [review]
one question inline, looks good otherwise

File: pkg/node/node_labels.go (hunk 1 above)
Comment (pchaigno) [reply pkg/node/node_labels.go]: Ah yeah, that makes more sense.

I named it after `node_address_xxx.go`. Should I rename those as well?

Comment (aanm) [reply pkg/node/node_labels.go]: I guess it would make sense yes.

File: pkg/endpointmanager/manager.go (hunk 2 above)
Comment (pchaigno) [reply pkg/endpointmanager/manager.go:482]: I don't think we need it. `node.GetLabels()` will return an uninitialized variable in that case; I have checked that it works fine by removing the `node.SetLabels(n.Labels)` statement below. The `Runtime*` tests are also passing so it should be okay.

**Comment by pchaigno** [comment]
test-me-please

**Review by christarazi — APPROVED** [review]

**Comment by pchaigno** [comment]
test-focus K8sServicesTest Checks service across nodes Tests NodePort BPF Tests with vxlan Tests NodePort with sessionAffinity from outside

**Comment by pchaigno** [comment]
I think this is ready to merge. I filled #11698 for the test failing on `K8s-1.11-Kernel-netnext`. I don't see how it could be caused by this PR and it happened in other PRs as well according to the CI dashboard and logs. All other required builds are passing.

**Comment by christarazi** [comment]
Great, merging.

**Pull request merged**